src/components/VerifyEmail/verify-email.tsx

```diff
@@ -0,0 +1,115 @@
+import { sendSignInLinkToEmail } from 'firebase/auth';
+import React from 'react';
+import { useSearchParams } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { Button } from '../../lib/Button';
+import { openToast } from '../../lib/Toast';
+import { firebaseAuth } from '../../utils/firebase';
+
+function VerifyEmailPage() {
+  const [query] = useSearchParams();
+
+  const handleResendEmail = async () => {
+    const accountRequiredButNotThere = !query.get('accountId') && query.get('isRecovery') !== 'true';
+    if (
+      accountRequiredButNotThere
+      || !query.get('email')
+      || !query.get('email').length
+      || !query.get('publicKey')
+      || !query.get('publicKey').length
+    ) return;
+
+    const accountId = query.get('accountId');
+    const publicKey = query.get('publicKey');
+    const email = query.get('email');
+    const isRecovery = query.get('isRecovery');
+    const success_url = query.get('success_url');
+    const failure_url = query.get('failure_url');
```

Comment:
See earlier comment about making this API only have a single URL, but with a `result` and optional `reason` for errors.

Also, can we get this extraction logic into a dedicated hook so that if we add or remove keys to/from this payload, we only need to do it in one place